cmd/snap-recovery/partition/mkfs.go

```diff
@@ -24,10 +24,10 @@ import (
 	"github.com/snapcore/snapd/gadget"
 )
 
-func MakeFilesystems(created map[string]gadget.LaidOutStructure) error {
-	for node, part := range created {
+func MakeFilesystems(created []deviceStructure) error {
```

Comment:
Having a public function that takes a private argument is difficult to use externally.

Reply:
Indeed, I will fix in a followup, I expect some tweaks here anyway :)

Reply:
And indeed it did change in the followup 7640